src/saltext/vmware/modules/esxi.py

```diff
@@ -1166,6 +1166,246 @@ def get_user(
         raise salt.exceptions.SaltException(str(exc))
 
 
+def add_role(
+    role_name,
+    privilege_ids,
```

Comment:
Why aren't we applying roles to multiple hosts like most other functions?

Reply:
The AuthorizationManager attribute is available only on the service instance and not at per host level like we did it for users. So, the roles need to be set at every host (pyvmomi shortcoming).

Reply:
I think some documentiation could help clear this up.

"Add local role on ESXi host" makes it sound like it can't apply to the entire vCenter instance.

"Add local role to service instance, which may be an ESXi host or vCenter instance" may be more clear?

Reply:
ok will change